course/src/main/java/in/testpress/course/adapter/BaseContentListAdapter.kt

```diff
@@ -0,0 +1,83 @@
+package `in`.testpress.course.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import `in`.testpress.core.TestpressSdk
+import `in`.testpress.course.R
+import `in`.testpress.course.databinding.RunningUpcomingListItemBinding
+import `in`.testpress.course.domain.DomainContent
+import `in`.testpress.course.domain.asDomainContent
+import `in`.testpress.util.ViewUtils
+
+abstract class BaseContentListAdapter<T : Any>(COMPARATOR: DiffUtil.ItemCallback<T>):
+    PagingDataAdapter<T, BaseContentListViewHolder>(COMPARATOR){
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseContentListViewHolder {
+        val binding = RunningUpcomingListItemBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+        return BaseContentListViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: BaseContentListViewHolder, position: Int) {
+        val content = getItem(position)
+        if (content != null) {
+            val domainContent = content.asDomainContent()
+            holder.bind(domainContent) { onItemClick(domainContent, holder.itemView.context) }
+            holder.setDateAndVisiblity(
```

Comment:
This name itself is confusing. If I am going to set a date then obviously it should be visible.